src/ssl.c

```diff
@@ -33904,6 +33904,22 @@ void wolfSSL_get0_next_proto_negotiated(const WOLFSSL *s, const unsigned char **
 
 #endif /* WOLFSSL_NGINX  / WOLFSSL_HAPROXY */
 
+#ifdef OPENSSL_EXTRA
+int wolfSSL_CTX_curve_is_disabled(WOLFSSL_CTX* ctx, word16 curve_id)
+{
+    return (curve_id <= WOLFSSL_ECC_MAX &&
+            ctx->disabledCurves &&
+            ctx->disabledCurves & (1 << curve_id));
+}
+
+int wolfSSL_curve_is_disabled(WOLFSSL* ssl, word16 curve_id)
+{
+    /* FIXME: see wolfSSL_set1_curves_list() below on why
```

Comment:
To fix this the following should be adding to struct WOLFSSL:

```
#ifdef OPENSSL_EXTRA
word32            disabledCurves;   /* curves disabled by user */
#endif
```

And in `InitSSL` `ssl-> disabledCurves = ctx->disabledCurves`.

Then refactor `wolfSSL_CTX_curve_is_disabled` to either take `disabledCurves` directly or both WOLFSSL_CTX and WOLFSSL and use ssl if populated otherwise use ctx.

Reply:
I'd be happy to.

Reply:
Added in the last push.